### PR TITLE
chore(hobby deployments): various fixes

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Problem
Fix hobby deployments as they are currently broken for all new installations since #9847 has been merged. Error:
```
Service clickhouse is in version 22.3.6. Expected range: >=21.6.0,<21.12.0. PostHog may not work correctly with the current version. To continue anyway, add SKIP_SERVICE_VERSION_REQUIREMENTS=1 as an environment variable
```

This is happening because we've bumped ClickHouse version while we are still referencing `posthog/posthog:latest-release` as the container image to use.

## Changes

I'm adding an optional override to be able to specify the `POSTHOG_APP_TAG` (aka: PostHog image tag) during the installation. I didn't find an easy way to fix this for everyone without either changing the default `POSTHOG_APP_TAG` or downgrade ClickHouse. The next release should fix this issue. 

## How did you test this code?
Tested on a new VM running the script from this branch:

```
curl -s -o deploy-hobby.sh https://raw.githubusercontent.com/posthog/posthog/fix_hobby_again/bin/deploy-hobby
export POSTHOG_APP_TAG=latest
bash deploy-hobby.sh
```

### CODE REVIEW

File: bin/deploy-hobby [hunk 1]
```diff
@@ -31,7 +33,7 @@ echo ""
 echo "Do you have a Sentry DSN you would like for debugging should something go wrong?"
 echo "If you do enter it now, otherwise just hit enter to continue"
 read -r SENTRY_DSN
-export SENTRY_DSN="${SENTRY_DSN:-'https://public@sentry.example.com/1'}" $SENTRY_DSN
```
Comment (guidoiaquinti) [inline bin/deploy-hobby:34]: I don't get why this were here but I might be missing something

**Review by hazzadous — APPROVED** [review]
lgtm

aside: Would it we possible to drop down to 21.11 or would that introduce risk of downgrade issues?

File: bin/deploy-hobby (hunk 1 above)
Comment (fuziontech) [reply bin/deploy-hobby:34]: Why is there a sentry DSN here? Great question. Basically want to make it a bit easier to debug the stack should the user want to. This is kind of similar to what CH does https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings/#server_configuration_parameters-send_crash_reports

We could reset this to an endpoint that we control by default to catch errors quicker honestly, but that seems a bit questionable from data collection POV.

**Comment by guidoiaquinti** [comment]
> aside: Would it we possible to drop down to 21.11 or would that introduce risk of downgrade issues?

We can but we will risk downgrade issues. Considering that by EOW we should cut a new release I'm more inclined to keep this like this. What do you think?

**Comment by hazzadous** [comment]
> What do you think?

Given these shouldn't be production instances I would just reduce the version. However given people are using in production then I'm ok with this and we use `POSTHOG_APP_TAG` to handle issues in support, was that the intention?

**Comment by guidoiaquinti** [comment]
> > What do you think?
> 
> Given these shouldn't be production instances I would just reduce the version. However given people are using in production then I'm ok with this and we use `POSTHOG_APP_TAG` to handle issues in support, was that the intention?

Yes, this is a fix for a [customer support request](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1653314179556439).

**Comment by hazzadous** [comment]
> Yes, this is a fix for a [customer support request](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1653314179556439).

LGTM

File: bin/upgrade-hobby [hunk 2]
```diff
@@ -10,7 +12,7 @@ else
 fi
 
 [[ -f ".env" ]] && export $(cat .env | xargs) || ( echo "No .env file found. Please create it with POSTHOG_SECRET and DOMAIN set." && exit 1)
-export POSTHOG_APP_TAG="${POSTHOG_APP_TAG:-latest-release}" $POSTHOG_APP_TAG
```
[the remark this answers is not in the document]
Comment (fuziontech) [reply bin/upgrade-hobby:13]: This was added to make it easier to test new releases

File: bin/deploy-hobby (hunk 1 above)
Comment (hazzadous) [reply bin/deploy-hobby:34]: I think it's in reference to the extra `$SENTRY_DSN` rather than why it's there?

Comment (hazzadous) [reply bin/deploy-hobby:34]: i.e. it works but it's also going to be exporting a var with a name of whatever `$SENTRY_DSN` evaluates to

Comment (guidoiaquinti) [reply bin/deploy-hobby:34]: My question was about the 2nd `$SENTRY_DSN` as I didn't get why it was needed

**Comment by fuziontech** [comment]
Something else we can do is just expose and set `SKIP_SERVICE_VERSION_REQUIREMENTS` env var for now. This will skip that version check (we know 22.3 works for the current release).

**Comment by hazzadous** [comment]
> Something else we can do is just expose and set `SKIP_SERVICE_VERSION_REQUIREMENTS` env var for now. This will skip that version check (we know 22.3 works for the current release).

ah right. I know there were some tests failing previously, but were they just test changes that were made?

**Comment by fuziontech** [comment]
Correct, they were broken tests - nothing relating to 22.3

EDIT: Let me double check by doing a quick deploy with last release though

**Review by fuziontech — APPROVED** [review]
lgtm

**Pull request merged**